src/views/templates/actions/VirtualMachineTemplatesActions.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Action } from '@openshift-console/dynamic-plugin-sdk';
+import { Dropdown, DropdownItem, DropdownPosition, KebabToggle } from '@patternfly/react-core';
+
+import useVirtualMachineTemplatesActions from './hooks/useVirtualMachineTemplatesActions';
+
+type VirtualMachineTemplatesActionsProps = { template: V1Template };
+
+const VirtualMachineTemplatesActions: React.FC<VirtualMachineTemplatesActionsProps> = ({
+  template,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const actions = useVirtualMachineTemplatesActions(template);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const handleClick = (action: Action) => {
+    if (typeof action?.cta === 'function') {
+      action?.cta();
+      setIsOpen(false);
+    }
+  };
+
+  const disabledActionInfo = (isActionDisabled: boolean, actionId: string): string => {
+    if (isActionDisabled) {
+      switch (actionId) {
+        case 'delete-template':
+          return t('Common templates cannot be deleted');
+        default:
+          return t('This action is currently not avaliable');
+      }
+    }
+    return '';
+  };
+
+  return (
+    // TODO: use LazyActionMenu when fixed
+    <Dropdown
+      isPlain
+      isOpen={isOpen}
+      position={DropdownPosition.right}
+      toggle={<KebabToggle onToggle={setIsOpen} />}
+      dropdownItems={actions?.map((action) => (
+        <DropdownItem
+          key={action?.id}
+          onClick={() => handleClick(action)}
+          isDisabled={action?.disabled}
+          description={disabledActionInfo(action?.disabled, action?.id)}
```

Comment:
description should not change, we can fix it in a followup.
```suggestion
          description={t(action?.description)}
```